Added OAI RAN tutorial (#14)
open-5g/openrangym: _posts/tutorials/2024-04-24-oai-tutorial.md

---
layout: post
title: "Deploying OpenAirInterface (OAI) 5G RAN and Core-network in Colosseum"
date: 2024-04-24
category: tutorials
author:
short-description: How to deploy the latest version of OAI gNB, UE and 5G Core network in Colosseum and establish end-to-end connection
---

> There are readymade images in Colosseum NAS that contains OAI RAN and 5G Core-network. These images can be selected in the Colosseum reservation page under `common` section. The images of interest are the following

Reviewer: ```suggestion
There are ready-made images in Colosseum NAS which include OAI RAN and 5G Core Network. These images can be selected in the Colosseum reservation page under the `common` section. The images of interest are
```

Reviewer 2: ```suggestion
There are ready-made images in Colosseum, which include OAI RAN and 5G Core Network. These images can be selected in the `common` image section of the Colosseum reservation page. The images of interest are
```

 - RAN Image: `oai-ran-2024-w13`
 - Core-network Image: `oai-5gcore-2024`

Make a reservation with `n+2` SRNs out of which one Core-network image and `n+1` RAN Images where `n` is the number of UEs in the test scenario.

Once the reservation is active, you can login to the SRNs. The password of `root` user is `pass`.

## Updating RAN Image

> The OAI version in the RAN image can be updated to the current latest one by following the below mentioned instructions. This step optional and required only when needed to update or change the OAI to a specific `develop` version.

Reviewer: ```suggestion
The OAI version in the RAN image can be updated to the latest version by following the instructions below. This step is optional and only required if you need to update or change the OAI to a specific `develop` version.
```

```
./enable_net.sh #set route and DNS
cd openairinterface5g/ #go to OAI directory
git fetch #fetch from repository
git checkout <tag or commit hash> #checkout to a specific version
source oaienv
cd cmake_targets/
./build_oai --gNB --nrUE -w USRP --ninja
```
## Run OAI 5G Core-network
In SRN running `oai-5gcore-2024`
```
cd oai-cn5g
docker compose up -d
```
This should start all the required docker containers for running the core-network.
## Run OAI gNB

> The image contains a python tool to start the gNB  and UE.

Reviewer: ```suggestion
The image contains a python script to start the gNB and UE.
```

```

> cd OAI-Colosseum\

Reviewer 2: ```suggestion
cd OAI-Colosseum
```

python3 ran.py -t donor -m sa
```

> This will start the gNB process. If you encounter an error related to FPGA compatibility number mismatch, then run the above command with `-f` option. This will flash the FPGA in USRP with the version that matches the UHD driver. Hence, this option is required only once. The subsequent runs can be started without `-f`.

Reviewer: ```suggestion
This will start the gNB process. If you get an error related to FPGA compatibility number mismatch, then run the above command with `-f` option. This will flash the FPGA in USRP with the version that matches the UHD driver. This option is only needed once. The subsequent runs can be started without `-f`.
```

Reviewer 2: ```suggestion
This will start the gNB process. If you get an error related to FPGA compatibility number mismatch, run the above command with `-f` option. This will flash the FPGA of the USRP with the version that matches your version of the UHD driver. This option is only needed once. The subsequent runs can be started without `-f`.
```

## Start RF Scenario

> Now, we need to tell the MCHEM  (Massive CHannel EMulator) to setup a channel between all the nodes in the current reservation. This can be done from any one of the SRN in the reservation. The following command start a RF scenario with 0 dB path loss.

Reviewer: ```suggestion
Now we need to tell the MCHEM (Massive CHannel EMulator) to set up a channel between all the nodes in the current reservation. This can be done from any of the SRNs in the reservation. The following command starts an RF scenario with 0 dB path loss.
```

Reviewer 2: ```suggestion
Now we need to tell the MCHEM (Massive CHannel EMulator) to set up a channel between all the nodes in the current reservation. This can be done from any of the SRNs in the reservation. The following command starts an RF scenario with 0 dB path loss. (Please note that losses due to hardware impairments are still present.)
```

`colosseumcli rf start -c 10011`

## Run OAI UE
Run the following commands.
```

> cd OAI-Colosseum\

Reviewer 2: ```suggestion
cd OAI-Colosseum
```

python3 ran.py -t ue -m sa
```

> If everything goes well, then the UE should get connected to the gNB and the core-network should have assigned an IP address to the UE. This can be verified by running `ifconfig` in UE side and check for the network interface named `oaitun_ue` has IP assigned.

Reviewer: ```suggestion
If all goes well, the UE should be connected to the gNB and the core network should have assigned an IP address to the UE. This can be verified by running `ifconfig` on the UE side and checking if the network interface named `oaitun_ue` has an IP assigned.
```

Reviewer 2: ```suggestion
If all goes well, the UE should connect to the gNB and the core network should have assigned an IP address to the UE. This can be verified by running `ifconfig` on the UE side and checking if the network interface named `oaitun_ue` has an IP assigned.
```

> Sometimes, the UE might continuously fail the Random Access (RA) procedure and would not proceed beyond that point. If that happens, try changing (increase or decrease with steps of 5) the `timing_advance` value  under the `106` PRB section in `conf.json` file.

Reviewer: ```suggestion
Sometimes the UE may continuously fail the Random Access (RA) procedure and not proceed beyond that point. If this happens, try changing the `timing_advance` value in the `106` PRB section of the `conf.json` file (increase or decrease in steps of 5).
```

## IP Traffic

> Ping the core-network from UE using `ping 192.168.70.135 -I oaitun_ue`. If the ping is successful, then we confirm that there is end-to-end connection between UE and 5G core-network. The user can push more traffic via the RAN using `iperf3` tool.

Reviewer: ```suggestion
Ping the core-network from UE using `ping 192.168.70.135 -I oaitun_ue`. If the ping is successful, it means an end-to-end connection between the UE and the 5G core network. has been established. 
The user can push more traffic over the RAN using `iperf3` tool.
```
